feat(typeahead): apply angular watch to the typeahead-editable setting

damonf opened the pull request:
Adds a watch for the typeahead-editable attribute.

icfantv commented:
@damonf, what problem is this PR solving?  Also, you're missing functional tests.

Please give [this](https://github.com/angular-ui/bootstrap/blob/master/CONTRIBUTING.md) a read and in the future, it's generally a good idea to run an idea by the team first before spending time on a PR as we don't want you to possibly be wasting your time.  Thanks.

wesleycho commented:
There was a prior issue reported by @damonf related to this in #4592.

wesleycho commented on src/typeahead/typeahead.js:
```diff
@@ -44,6 +44,10 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.debounce', 'ui.bootstrap
     var isEditable = originalScope.$eval(attrs.typeaheadEditable) !== false;
+    originalScope.$watch(attrs.typeaheadEditable, function (newVal) {
+      isEditable = newVal !== false;
+      modelCtrl.$setValidity('editable', isEditable);
```
This is a breaking change, and editable state should not be related to ngModel's validity. This is a bad change.

wesleycho commented:
This needs tests.

The model $setValidity call needs to be removed as well.

damonf commented:
Thanks for your patience.
So if i removed the $setValidity call and added some unit tests you would
accept the PR?

On Thu, Nov 5, 2015 at 5:54 PM, Wesley Cho <notifications@github.com> wrote:

> This needs tests.
>
> The model $setValidity call needs to be removed as well.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular-ui/bootstrap/pull/4820#issuecomment-153972976>
> .
>

wesleycho commented:
I would be willing to accept this feature with the necessary changes.
